src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -645,15 +645,15 @@ discover_dfuse(void)
 			dfs_list[num_dfs].dfs_dir_hash = NULL;
 			dfs_list[num_dfs].len_fs_root  = strnlen(fs_entry->mnt_dir, DFS_MAX_PATH);
```

Comment:
In general dfs_list[num_dfs] could be left partially initialised by this function.

Reply:
@ashleypittman Could you please be more clear? In case of "partially initialized", it should be fine since "num_dfs" is not increased, right?

Reply:
True, it would be partially initialised but then re-used and likely overwritten later.

Reply:
Right. Such scenario looks fine to me. Do you see any issue about this? Thank you!

Reply:
it's probably not a visible issue but it does depend on other users of this data structre being correct.  It would be much safer to dynamically allocate these and search for them in a hash table.

Reply:
Current static array is simple to use and convenient in debugging. We can change it as you suggested later if we do need to handle large number of DFS mount points in future. Thank you!